File: DotNET/Endpoint Examples/Multipart Payload/delete-resource.cs
/*
 * What this sample does:
 * - Deletes a resource by id.
 * - Routed from Program.cs as: `dotnet run -- delete-resource-multipart <id>`.
 *
 * Setup (environment):
 * - Copy .env.example to .env
 * - Set PDFREST_API_KEY=your_api_key_here
 * - Optional: set PDFREST_URL to override the API region. For EU/GDPR compliance and proximity, use:
 *     PDFREST_URL=https://eu-api.pdfrest.com
 *   For more information visit https://pdfrest.com/pricing#how-do-eu-gdpr-api-calls-work
 */

namespace Samples.EndpointExamples.MultipartPayload
{
    public static class DeleteResource
    {
        public static async Task Execute(string[] args)
        {
            if (args == null || args.Length < 1)
            {
                Console.Error.WriteLine("delete-resource-multipart requires <id>");
                Environment.Exit(1);
                return;
            }

            var id = args[0];
            var apiKey = Environment.GetEnvironmentVariable("PDFREST_API_KEY");
            if (string.IsNullOrWhiteSpace(apiKey))
            {
                Console.Error.WriteLine("Missing required environment variable: PDFREST_API_KEY");
                Environment.Exit(1);
                return;
            }
            var baseUrl = Environment.GetEnvironmentVariable("PDFREST_URL") ?? "https://api.pdfrest.com";
            var url = baseUrl.TrimEnd('/') + "/resource/" + id;

            using var client = new HttpClient();
            using var request = new HttpRequestMessage(HttpMethod.Delete, url);
            request.Headers.TryAddWithoutValidation("Api-Key", apiKey);
            var response = await client.SendAsync(request);
            var body = await response.Content.ReadAsStringAsync();
            Console.WriteLine(body);
        }
    }
}

File: PHP/Endpoint Examples/Multipart Payload/delete-resource.php
<?php

// By default, we use the US-based API service. This is the primary endpoint for global use.
$apiUrl = "https://api.pdfrest.com";

/* For GDPR compliance and enhanced performance for European users, you can switch to the EU-based service by uncommenting the URL below.
 * For more information visit https://pdfrest.com/pricing#how-do-eu-gdpr-api-calls-work
 */
//$apiUrl = "https://eu-api.pdfrest.com";

$curl = curl_init();

curl_setopt_array($curl, array(
  CURLOPT_URL => $apiUrl.'/resource/xxxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx',
  CURLOPT_RETURNTRANSFER => true,
  CURLOPT_ENCODING => '',
  CURLOPT_MAXREDIRS => 10,
  CURLOPT_TIMEOUT => 0,
  CURLOPT_FOLLOWLOCATION => true,
  CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
  CURLOPT_CUSTOMREQUEST => 'DELETE',
  CURLOPT_HTTPHEADER => array(
    'api-key: xxxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'
  ),
));

$response = curl_exec($curl);

curl_close($curl);

echo $response;
